"""empty message

Revision ID: bf6336472889
Revises: ff07799f0639
Create Date: 2021-09-23 12:36:56.239842

"""

# revision identifiers, used by Alembic.
revision = 'bf6336472889'
down_revision = 'ff07799f0639'

from alembic import op
import sqlalchemy as sa


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('ibm_clouds',
    sa.Column('id', sa.String(length=32), nullable=False),
    sa.Column('name', sa.String(length=255), nullable=False),
    sa.Column('api_key', sa.String(length=500), nullable=False),
    sa.Column('api_key_id', sa.String(length=500), nullable=False),
    sa.Column('status', sa.Enum('AUTHENTICATING', 'VALID', 'INVALID', 'ERROR_DUPLICATE_RESOURCE_GROUPS'), nullable=False),
    sa.Column('user_id', sa.String(length=32), nullable=False),
    sa.Column('project_id', sa.String(length=32), nullable=False),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('ibm_credentials',
    sa.Column('id', sa.String(length=32), nullable=False),
    sa.Column('access_token', sa.Text(), nullable=False),
    sa.Column('refresh_token', sa.Text(), nullable=True),
    sa.Column('expiration_date', sa.DateTime(), nullable=True),
    sa.Column('cloud_id', sa.String(length=32), nullable=False),
    sa.ForeignKeyConstraint(['cloud_id'], ['ibm_clouds.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('ibm_service_credentials',
    sa.Column('id', sa.String(length=32), nullable=False),
    sa.Column('resource_instance_id', sa.String(length=1000), nullable=False),
    sa.Column('cloud_id', sa.String(length=32), nullable=False),
    sa.ForeignKeyConstraint(['cloud_id'], ['ibm_clouds.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('ibm_service_credentials')
    op.drop_table('ibm_credentials')
    op.drop_table('ibm_clouds')
    # ### end Alembic commands ###